client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,380 @@
+			return fmt.Errorf("error starting CPU profiler: %w", err)
+		}
+		err = pprof.StartCPUProfile(f)
+		if err != nil {
+			return fmt.Errorf("error starting CPU profiler: %w", err)
+		}
+		defer pprof.StopCPUProfile()
+	}
+
+	defer func() {
+		if pv := recover(); pv != nil {
+			log.Criticalf("Uh-oh! \n\nPanic:\n\n%v\n\nStack:\n\n%v\n\n",
+				pv, string(debug.Stack()))
+		}
+	}()
+
+	// Prepare the Core.
+	clientCore, err := dexCore.New(cfg.Core(logMaker.Logger("CORE")))
+	if err != nil {
+		return fmt.Errorf("error creating client core: %w", err)
+	}
+
+	// Handle shutdown by user (if running via terminal), or on system shutdown.
+	// TODO: SIGTERM is apparently spoofed by Go for Windows. Nice feature, but
+	// not well documented. Test to verify. Could also catch SIGKILL, which is
+	// sent after a configured timeout if the program doesn't exit on SIGTERM.
+	killChan := make(chan os.Signal, 1)
+	signal.Notify(killChan, os.Interrupt /* ctrl-c */, syscall.SIGTERM /* system shutdown */, syscall.SIGQUIT /* ctrl-\ */)
+	go func() {
+		for range killChan {
+			log.Infof("Shutting down...")
+			cancel()
+			return
+		}
+	}()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		clientCore.Run(appCtx)
+		cancel() // in the event that Run returns prematurely prior to context cancellation
+	}()
+
+	<-clientCore.Ready()
+
+	defer func() {
+		log.Info("Exiting dexc main.")
+		cancel()  // no-op with clean rpc/web server setup
+		wg.Wait() // no-op with clean setup and shutdown
+	}()
+
+	var marketMaker *mm.MarketMaker
+	if cfg.Experimental {
+		// TODO: on shutdown, stop market making and wait for trades to be
+		// canceled.
+		marketMaker, err = mm.NewMarketMaker(clientCore, logMaker.Logger("MM"))
+		if err != nil {
+			return fmt.Errorf("error creating market maker: %w", err)
+		}
+	}
+
+	if cfg.RPCOn {
+		rpcSrv, err := rpcserver.New(cfg.RPC(clientCore, marketMaker, logMaker.Logger("RPC")))
+		if err != nil {
+			return fmt.Errorf("failed to create rpc server: %w", err)
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			cm := dex.NewConnectionMaster(rpcSrv)
+			err := cm.Connect(appCtx)
+			if err != nil {
+				log.Errorf("Error starting rpc server: %v", err)
+				cancel()
+				return
+			}
+			cm.Wait()
+		}()
+	}
+
+	webSrv, err := webserver.New(cfg.Web(clientCore, logMaker.Logger("WEB"), utc))
+	if err != nil {
+		return fmt.Errorf("failed creating web server: %w", err)
+	}
+
+	webStart := make(chan error)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		cm := dex.NewConnectionMaster(webSrv)
+		webStart <- cm.Connect(appCtx)
+		cm.Wait()
+	}()
+
+	if err := <-webStart; err != nil {
+		return err
+	}
+
+	// No errors running webserver, so we can be certain we won any race between
+	// starting instances. Start the sync server now.
+	openC := make(chan struct{}) // no buffer. Is ignored if window is currently open
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		runServer(appCtx, syncDir, openC, killChan, cfg.Net)
```

Comment:
It doesn't look like the approach with macdrive requires the "syncserver" anymore, correct?
If so, can remove this and put `//go:build !darwin` in syncserver.go

Reply:
Well, the sycnserver is still useful regardless. New windows will be created if the user attempts to start dexc-desktop from cmd if it's already running.

Reply:
This is a messy hack to prevent multiple instances.  The Mac binary should have a mechanism for it to be a singleton app.  I can't believe everyone has to invent something like this to accomplish a very common requirement.

Reply:
According to this, Mac apps are inherently single instance: https://stackoverflow.com/a/1081255
It's that no longer accurate, or is there a setting?

Reply:
Even if we need a solution for when people try to launch it repeatedly from the command line, a better solution is one that does not require a promiscuous web server. With the multiple window limitation gone in macdrive, we only need to detect a running instance and stop it from launching another.
A named resource or some kind of file lock would be preferable, even if we address that in future work.

Reply:
> This is a messy hack to prevent multiple instances

np, we can do without the syncserver on mac.